packages/account-template/src/accountLinks.ts

```diff
@@ -0,0 +1,95 @@
+import type { CeramicApi, CeramicDocument } from './types'
+
+import { createLink } from '3id-blockchain-utils'
+
+import { callRpc } from './utils' 
+
+class AccountLinks {
+  private _accountLinkDocuments: Record<string, CeramicDocument>;
+
+  constructor (public ceramicDoc: CeramicDocument, private _ceramic: CeramicApi, public provider?: any) {
+    this._accountLinkDocuments = {}
+    this.ceramicDoc.on('change', this._loadAccountLinkDocs.bind(this))
+  }
+
+  getLinkedAddresses (): Array<string> {
+    return Object.keys(this._accountLinkDocuments)
+  }
+
+  async linkAddress (address: string, proof?: Record<string, any>): Promise<void> {
+    if (!proof) {
+      if (!this.provider) {
+        throw new Error('Provider must be set')
```

Comment:
If provider is not set and we get an ethereum address we should assume `eip155:1`